packages/ddp-client/common/livedata_connection.js

```diff
@@ -890,14 +890,18 @@ export class Connection {
     // If we're using the default callback on the server,
     // block waiting for the result.
     if (future) {
+      future.stub = stubReturnValue;
+      future.serverResult = future;
```

Comment:
I would suggest renaming these to `stubPromise` and `serverPromise` for consistency, and to make it clear what the properties contain.

Reply:
As far as I can tell, `stubReturnValue` is a value and not a promise that is passed into `_apply` from `applyAsync`. I think the code needs to be changed upstream of `future.stubPromise = stubReturnValue` in order for things to work as expected. I could be wrong though. There's a lot going on in this code :) 

https://github.com/meteor/meteor/blob/fd2150c4c2d41a6a625b54cf6bad41598f8a6176/packages/ddp-client/common/livedata_connection.js#L721

Also it's not clear to me why `stubInvocation` is being awaited here when it is not a promise:
https://github.com/meteor/meteor/blob/fd2150c4c2d41a6a625b54cf6bad41598f8a6176/packages/ddp-client/common/livedata_connection.js#L972-L982